### ISSUE TITLE
GPU-less install

### ISSUE THREAD

**Issue by ChrisRackauckas** [JuliaPlots/Makie.jl#56]
One of the great things about Plots.jl's setup is it can install pretty much anywhere because the extra functionality is all provided by backends. I can't see getting Makie.jl to work on my friendly neighborhood cluster, while GR works fine. And then there are cases where there's no access to OpenGL (without VirtualGL), but other options like Plotly works.

Will there be a nice way to be able to install Makie in those setups? I like the GLVisualize-first approach and think it makes a lot of sense to ensure that there's a fully complete and fully Julia backend, but at the same time it limits the usefulness if it has to be directly required.

**Comment by SimonDanisch** [comment]
@jpsamaroo wanted to help me get a backend without a gpu running :) So that might work in the near future!

**Comment by jpsamaroo** [comment]
Indeed, I'm planning on setting up a Cairo+Gtk backend for Makie (Gtk windowing stuff should be optional, to support headless systems). That way we can at least support 2D plotting, and then later on consider adding support for OpenGL for 3D plots (which could be provided by VirtualGL or llvmpipe on systems without proper graphics hardware).

**Comment by gdkrmr** [comment]
Could this work with a software renderer? https://stackoverflow.com/questions/7310885/opengl-without-a-graphics-card

Or maybe virtualgl?

**Comment by mkborregaard** [comment]
There is CairoMakie now.

**Comment by gdkrmr** [comment]
On a headless system without graphics card I installed software rendering and `glxgears` works fine.
I also installed `Makie` and it works fine if I use `ssh -Y server`.

Using `xpra` you can launch a virtual x server on the headless system and use that (https://help.ubuntu.com/community/Xpra) so you don't need to use `ssh -Y`

There was one warning:
```
Warning: GLFW.GLFWError(PLATFORM_ERROR::ErrorCode = 65544, "X11: RandR gamma ramp support seems broken")
```

You can even use a local `xpra` and open the display on the remote server, but mouse interaction will produce tons of error messages on the remote julia console.

Next step is to get it working inside Jupyter notebooks... I will let you know if I manage to do that.

**Comment by SimonDanisch** [comment]
Pretty cool you got that running - and it doesn't even sound that complicated ;)

**Comment by gdkrmr** [comment]
It was not that complicated. How do I resize the window opened by Makie from within Julia? My current issue is that the window opened in xpra is tiny and therefore it is saved as a tiny png.

**Comment by asinghvi17** [comment]
You could try passing a resolution to the Scene?  For example, `sc = Scene(resolution = (1920, 1080))`

**Comment by gdkrmr** [comment]
```julia
p = Scene(resolution = (1920, 1080))
volume!(p, rand(32, 32, 32))
save("make_test.png", p)
```
works on my machine, but the png is still tiny on the headless machine.

**Comment by SimonDanisch** [comment]
Hm, the only reason it would ignore this, if it somehow thinks that Juno is there!
https://github.com/JuliaPlots/AbstractPlotting.jl/blob/master/src/display.jl#L59
If you use remote Atom, try disabling the plotpane!

**Comment by gdkrmr** [comment]
I am trying to make it work with jupyter notebooks, I am not using Atom. I also tried `resize!` and it does not work. Maybe it has something to do with the display size (the monitor is 0mm x 0mm):
```
me@server:~$ xrandr -d :7 -q
xrandr:Failed to get size of gamma for output default
Screen 0: minimum 320 x 175, current 5280 x 1920, maximum 8192 x 4096
default connected 5280x1920+0+0 0mm x 0mm

... a list with all possible resolutions

```
I tried changing the resolution with `xrandr` but without success.
